core/src/test/java/io/questdb/test/griffin/AsOfJoinTest.java

```diff
@@ -798,6 +902,174 @@ public void testLtJoinSequenceGapOnKey() throws Exception {
+                    "('AAPL', '2000')," +
+                    "('AAPL', '2001')," +
+                    "('AAPL', '2002')," +
+                    "('AAPL', '2003')," +
+                    "('AAPL', '2004')," +
+                    "('AAPL', '2005')"
+            );
+            compile("insert into tests VALUES " +
+                    "('QSTDB', '2003')," +
+                    "('QSTDB', '2004')," +
+                    "('QSTDB', '2005')," +
+                    "('QSTDB', '2006')," +
+                    "('QSTDB', '2007')," +
+                    "('QSTDB', '2008')"
+            );
+
+            String query = "SELECT * " +
+                    "FROM tests t0 " +
+                    "LT JOIN (" +
+                    "   SELECT * " +
+                    "   FROM tests t1 " +
+                    "   LT JOIN (" +
+                    "       SELECT * " +
+                    "       FROM tests t2 " +
+                    "       LT JOIN (" +
+                    "           SELECT * FROM tests t3" +
+                    "       ) on (Ticker)" +
+                    "   ) ON (Ticker)" +
+                    ") ON (Ticker)";
+            String expected = "Ticker\tts\tTicker1\tts1\tTicker11\tts11\tTicker111\tts111\n" +
+                    "AAPL\t2000-01-01T00:00:00.000000Z\t\t\t\t\t\t\n" +
+                    "AAPL\t2001-01-01T00:00:00.000000Z\tAAPL\t2000-01-01T00:00:00.000000Z\t\t\t\t\n" +
+                    "AAPL\t2002-01-01T00:00:00.000000Z\tAAPL\t2001-01-01T00:00:00.000000Z\tAAPL\t2000-01-01T00:00:00.000000Z\t\t\n" +
+                    "QSTDB\t2003-01-01T00:00:00.000000Z\t\t\t\t\t\t\n" +
+                    "AAPL\t2003-01-01T00:00:00.000000Z\tAAPL\t2002-01-01T00:00:00.000000Z\tAAPL\t2001-01-01T00:00:00.000000Z\tAAPL\t2000-01-01T00:00:00.000000Z\n" +
+                    "QSTDB\t2004-01-01T00:00:00.000000Z\tQSTDB\t2003-01-01T00:00:00.000000Z\t\t\t\t\n" +
+                    "AAPL\t2004-01-01T00:00:00.000000Z\tAAPL\t2003-01-01T00:00:00.000000Z\tAAPL\t2002-01-01T00:00:00.000000Z\tAAPL\t2001-01-01T00:00:00.000000Z\n" +
+                    "AAPL\t2005-01-01T00:00:00.000000Z\tAAPL\t2004-01-01T00:00:00.000000Z\tAAPL\t2003-01-01T00:00:00.000000Z\tAAPL\t2002-01-01T00:00:00.000000Z\n" +
+                    "QSTDB\t2005-01-01T00:00:00.000000Z\tQSTDB\t2004-01-01T00:00:00.000000Z\tQSTDB\t2003-01-01T00:00:00.000000Z\t\t\n" +
+                    "QSTDB\t2006-01-01T00:00:00.000000Z\tQSTDB\t2005-01-01T00:00:00.000000Z\tQSTDB\t2004-01-01T00:00:00.000000Z\tQSTDB\t2003-01-01T00:00:00.000000Z\n" +
+                    "QSTDB\t2007-01-01T00:00:00.000000Z\tQSTDB\t2006-01-01T00:00:00.000000Z\tQSTDB\t2005-01-01T00:00:00.000000Z\tQSTDB\t2004-01-01T00:00:00.000000Z\n" +
+                    "QSTDB\t2008-01-01T00:00:00.000000Z\tQSTDB\t2007-01-01T00:00:00.000000Z\tQSTDB\t2006-01-01T00:00:00.000000Z\tQSTDB\t2005-01-01T00:00:00.000000Z\n";
+            assertQuery(expected, query, "ts", false, true);
+        });
+    }
```

Comment:
I reckon it'd be good to add test(s) with :
- more than  one join key 
- synthetic symbol column in master as join key (e.g. `select rnd_symbol('a', 'b')  ticker, ts from tests ` )

Reply:
added `testLtJoinOnCompositeSymbolKey()`. 
will add a test with a synthetic join column in a master, that's a good idea!

Reply:
done